### PR TITLE
CDAP-4405 docs for DAG in pipeline

### PR DESCRIPTION
**Pull request by gsps1**
JIRA: https://issues.cask.co/browse/CDAP-4405

### CODE REVIEW

File: cdap-docs/cdap-apps/source/hydrator/creating.rst [hunk 1]
```diff
@@ -147,7 +147,127 @@ property replaces the ``schedule`` property of an ETL Batch Application.
...
+DAG in transform
+----------------
+
+ETLConfig support DAG in pipeline, which allows non-linear execution of pipeline stages.
+
+Example :
+pipeline reads from the stream ``purchaseStats``. It replicates the stream events to the table ``replicaTable``,
+while it only writes the userIds to ``usersTable`` when a user's purchase price is greater that $1000,
+this filtering logic is applied using the script ``spendingUsersScript``.
+
```
Comment (awholegunch) [inline cdap-docs/cdap-apps/source/hydrator/creating.rst]:
```
In this example, a pipeline reads from the stream ``purchaseStats``. It replicates the stream events
to the table ``replicaTable``, while at the same time it writes just the ``userIds`` to the ``usersTable`` 
when a user's purchase price is greater that 1000. This filtering logic is applied using the script
``spendingUsersScript``:

File: cdap-docs/cdap-apps/source/hydrator/creating.rst [hunk 2]
```diff
@@ -147,7 +147,127 @@ property replaces the ``schedule`` property of an ETL Batch Application.
 
 In the example code above, we will use a *ProjectionTransform* (a type of Transform) to drop certain
 columns in the incoming data. A *StreamSink* in the final step needs a data field property
-that it will use as the content for the data to be written. 
+that it will use as the content for the data to be written.
+
+DAG in transform
```
Comment (awholegunch) [inline cdap-docs/cdap-apps/source/hydrator/creating.rst]: I don't know what this title is referring to. Can we come up with an alternative? It should be in "Capital Case".

File: cdap-docs/cdap-apps/source/hydrator/creating.rst [hunk 3]
```diff
@@ -147,7 +147,127 @@ property replaces the ``schedule`` property of an ETL Batch Application.
...
+----------------
+
+ETLConfig support DAG in pipeline, which allows non-linear execution of pipeline stages.
```
Comment (awholegunch) [inline cdap-docs/cdap-apps/source/hydrator/creating.rst]: What is ETLConfig? It is not mentioned elsewhere (except as a variable) in the documentation. Did you mean:

```
ETL Applications support directed acyclic graphs in pipelines, which allows for the 
non-linear execution of pipeline stages.

File: cdap-docs/cdap-apps/source/hydrator/creating.rst (hunk 2 above)
Comment (awholegunch) [inline cdap-docs/cdap-apps/source/hydrator/creating.rst]: What about "Non-linear Execution in Pipelines"?

File: cdap-docs/cdap-apps/source/hydrator/creating.rst (hunk 1 above)
Comment (awholegunch) [inline cdap-docs/cdap-apps/source/hydrator/creating.rst]: Should we add something pointing to the "connections" section, as that is what is creating the non-linear execution?

**Comment by awholegunch** [comment]
Please see Albert's comments in https://github.com/caskdata/cdap/pull/4862 as to additional material required: https://github.com/caskdata/cdap/pull/4862#discussion-diff-49542776

**Comment by gsps1** [comment]
mistakenly pulled develop into this branch. ill reopen.

**Pull request closed without merging**